src/main/java/uk/gov/pay/api/service/PaymentSearchService.java

```diff
@@ -54,6 +79,46 @@ public Response doSearch(Account account, String reference, String email, String
         queryParams.put(PAGE, pageNumber);
         queryParams.put(DISPLAY_SIZE, displaySize);
         
-        return searchCardPayments.getSearchResponse(account, queryParams);
+        return getSearchResponse(account, queryParams);
+    }
+    
+    private Response getSearchResponse(Account account, Map<String, String> queryParams) {
+        queryParams.put("transactionType", "charge");
+
+        String url = connectorUriGenerator.chargesURIWithParams(account, queryParams);
+        Response connectorResponse = client
+                .target(url)
+                .request()
+                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                .get();
+        LOGGER.info("response from connector for transaction search: {}", connectorResponse);
+        if (connectorResponse.getStatus() == SC_OK) {
+            return processResponse(connectorResponse);
+        }
+        throw new SearchPaymentsException(connectorResponse);
+    }
+
+    private Response processResponse(Response connectorResponse) {
+        PaymentSearchResponse response;
+        try {
+            response = connectorResponse.readEntity(PaymentSearchResponse.class);
+        } catch (ProcessingException ex) {
+            throw new SearchPaymentsException(ex);
```

Comment:
I'm not sure we need to catch ProcessingException here. I don't see any exception mappers for SearchPaymentException so I believe if a ProcessingException occurs a 500 would be returned as well.

Reply:
Just moved all this code rather than changing any of it, but i'll make some improvements in response to your comments

Reply:
We do have a mapper, it's called SearchChargesExceptionMapper though 🤦‍♂

Reply:
🤦‍♂

Reply:
Hmm actually I agree - why should we catch an exception, it's an internal error so why not allow it to propagate and return a generic 500

Reply:
I've removed the try...catch

Reply:
I guess it matters as the following test fails:
```
17:42:47  [ERROR]   PaymentResourceSearchIT.searchPayments_errorIfConnectorResponseIsInvalid:525 JSON path [$.code] doesn't match.
17:42:47  Expected:
17:42:47  is "P0498"
17:42:47  Actual:
17:42:47  500
```
sorry about that :(

Reply:
ah damn it, i'll just put it back

Reply:
I've put back the exception handling just around the readEntity statement